deploy/deploy.sh

```diff
@@ -1,57 +1,100 @@
+
+echo_usage() {
+  echo; echo "USAGE"; echo "  ${0} [-t tag] [-b branch]"; echo
+  echo "Deploy the latest SuiteC code from a specified branch or tag."; echo
+  echo "Set environment variable DOCUMENT_ROOT to the Apache directory which will serve SuiteC static files."; echo
+}
+
+# If we have missing requirements then echo usage info and exit.
+[[ $# -gt 0 ]] || { echo_usage; exit 0; }
+[[ "${DOCUMENT_ROOT}" ]] || { echo; echo "[ERROR] 'DOCUMENT_ROOT' is undefined"; echo_usage; exit 1; }
+
+# Default remote repository
+git_remote="origin"
+
+while getopts "b:r:t:" arg; do
+  case ${arg} in
+    b)
+      git_branch="${OPTARG}"
+      ;;
+    r)
+      git_remote="${OPTARG}"
+      ;;
+    t)
+      git_tag="${OPTARG}"
+      ;;
+  esac
+done
+
+# Validation
+[[ "${git_tag}" || "${git_branch}" ]] || { echo; echo "[ERROR] You must specify branch or tag."; echo_usage; exit 1; }
+[[ "${git_tag}" && "${git_branch}" ]] && { echo; echo "[ERROR] Specify branch or tag but NOT both."; echo_usage; exit 1; }
 
-# Clear any local changes present in the branch
+echo; echo "WARNING! In two seconds we will clear local changes with git reset. Control-c to abort."; echo
+sleep 2.5
+
+# Next steps are recorded in time-stamped log file
+logger="tee -a $(date +"${base_directory}/logs/deploy_%Y-%m-%d-%H%M%S.log")"
+
+echo "Deploy SuiteC with command: ${0} ${*}" | ${logger}
+
+# Clear local changes
 git reset --hard HEAD
 
-# Work on a temporary branch
-git checkout -b tmp
+tmp_branch_name=$(date +"deploy_%Y-%m-%d-%H%M%S")
+echo "Create a local, temporary Git branch: ${tmp_branch_name}" | ${logger}
+# :TODO: Is a tmp branch necessary?
+git checkout -b "${tmp_branch_name}"
 
-# Get all the branches and tags from the remote
-git fetch $TARGET_REMOTE
-git fetch -t $TARGET_REMOTE
+# Learn about remote branches
+git fetch ${git_remote}
 
-# Delete the local copy of the target branch (if any) as the upstream branch might have been rebased
-git rev-parse --verify $TARGET_BRANCH > /dev/null 2>&1 && git branch -D $TARGET_BRANCH
+if [[ "${git_branch}" ]]; then
+  # Delete the local copy of the target branch (if any) as the upstream branch might have been rebased
+  git rev-parse --verify ${git_branch} > /dev/null 2>&1 && git branch -D ${git_branch}
+  echo "Begin Git checkout of remote branch: ${git_remote}/${git_branch}" | ${logger}
+  git checkout ${git_remote}/${git_branch} || { echo "[ERROR] Unknown Git branch: ${git_branch}"; echo; exit 1 }
+else
+  # Learn about remote tags
+  git fetch -t ${git_remote}
+  git checkout tags/${git_tag} || { echo "[ERROR] Unknown Git tag: ${git_tag}"; echo; exit 1 }
+fi
+
+echo "The Git checkout is complete. Next, install." | ${logger}
 
 # Check out the branch or tag. If a tag is being deployed, the git HEAD will point to a commit and
 # will end up in a "detached" state. As we shouldn't be committing on deployed code, this is considered OK
-git checkout $TARGET_REMOTE/$TARGET_BRANCH
-git branch -D tmp
+# :TODO: Is a tmp branch necessary?
```

Comment:
@johncrossman I've wondered the same thing. Worth a minor-priority JIRA to investigate?